applications/pelion_client/src/modules/net_ot.cpp

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <zephyr.h>
+#include <stdio.h>
+
+#include <net/openthread.h>
+#include <openthread/thread.h>
+#include <openthread/netdata.h>
+
+#include "net_event.h"
+
+#define MODULE net
+#include <caf/events/module_state_event.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_PELION_CLIENT_NET_LOG_LEVEL);
+
+
+static struct k_delayed_work connecting_work;
+static enum net_state net_state;
+
+
+static void format_address(char *buffer, size_t buffer_size, const uint8_t *addr_m8, size_t addr_size)
+{
+	if (addr_size == 0) {
+		buffer[0] = 0;
+		return;
+	}
+
+	size_t pos = 0;
+
+	for (size_t i = 0; i < addr_size; i++) {
+		int ret = snprintf(&buffer[pos], buffer_size - pos, "%.2x", addr_m8[i]);
+
+		if ((ret > 0) && ((size_t)ret < buffer_size - pos)) {
+			pos += ret;
+		} else {
+			break;
+		}
+	}
+}
+
+
+static bool check_neighbors(otInstance *instance)
+{
+	otNeighborInfoIterator iterator = OT_NEIGHBOR_INFO_ITERATOR_INIT;
+	otNeighborInfo info;
+
+	bool has_neighbors = false;
+
+	while (otThreadGetNextNeighborInfo(instance, &iterator, &info) == OT_ERROR_NONE) {
+		char addr_str[ARRAY_SIZE(info.mExtAddress.m8) * 2 + 1];
+
+		format_address(addr_str, sizeof(addr_str), info.mExtAddress.m8,
```

Comment:
Not a big deal, but there's a standard `inet_ntop()` function for binary->string conversion. Unless you _really_ prefer a custom IPv6 string format (w/o colons).

Reply:
I prefer to use standard display method as long as I can use it, thanks.

Reply:
I will have to know the inet type here. I don't know it it will always be ipv6. Let's leave it as is.